src/internal/packager/validate/requirements.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package validate provides validation functions for package operations
+package validate
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/zarf-dev/zarf/src/api/v1alpha1"
+	"github.com/zarf-dev/zarf/src/config"
+)
+
+// VersionRequirementsError is returned when operational requirements are not met
+type VersionRequirementsError struct {
+	Requirements   []v1alpha1.VersionRequirement
+	CurrentVersion string
+}
+
+func (e *VersionRequirementsError) Error() string {
```

Comment:
Non-blocking, more of a discussion item. There's an alternative path for this where we use a `NewVersionRequirementsError(...) VersionRequirementsError` constructor function, and calculate all of the necessary fields upfront. Then the error method can assemble the message string from the pre-calculated fields. I expect that this approach would be a bit more readable, but it is otherwise equivalent.